bindings/tpython/src/TPython.cxx

```diff
@@ -176,6 +176,10 @@ Bool_t TPython::Import(const char *mod_name)
 
    // allow finding to prevent creation of a python proxy for the C++ proxy
    Py_INCREF(mod);
+   if (!CPyCppyy::gThisModule) {
```

Comment:
Slowly getting to your PRs/issues, thanks @ferdymercury !

`CPyCppyy::gThisModule` is actually not the ROOT module, it is `libcppyy`. And if you get to this point it is because `Initialize()` did not fail in line 168. `Initialize` imports ROOT, which in turn imports libcppyy, so this should not be necessary.

Reply:
In theory yes, but this is what I see with this simple reproducer:

![image](https://user-images.githubusercontent.com/10653970/116551464-f6594000-a8c5-11eb-820e-a18e0e3dc685.png)

So maybe the initialize function has to be modified to prevent reaching this point?

Reply:
Yes, the check should be rather in the initialization function.

On the other hand, please note that `TPython::Import` is intended to import Python modules for which we would like to autogenerate C++ equivalents for their classes and methods via Cling. I think this is not what you intend to do here, you just want to import numpy. For that purpose, please use `TPython::Exec("import numpy")` (as you do in the main message of this issue).